source/elements/oneDPL/source/parallel_api/execution_policies.rst

```diff
@@ -180,5 +178,55 @@ as the template argument, otherwise unspecified.
 Return a policy object constructed from ``policy``, with a new kernel name provided as the template
 argument. If no policy object is provided, the new policy is constructed from ``dpcpp_default``.
 
+.. _exec-policy-type-trait:
+
+Execution Policy Type Trait
++++++++++++++++++++++++++++
+
+oneDPL provides type trait utilities to detect at compile time its execution policy types for the purpose of
+function overload resolution:
+
+.. code:: cpp
+
+  // Defined in <oneapi/dpl/execution>
+
+  namespace oneapi {
+    namespace dpl {
+
+      template <class T>
+      struct is_execution_policy { /*see below*/ };
+
+      template <class T>
+      inline constexpr bool is_execution_policy_v = oneapi::dpl::is_execution_policy<T>::value;
+
+      namespace execution {
+
+        template <class T>
+        struct is_execution_policy { /*see below*/ };
+
+        template <class T>
+        inline constexpr bool is_execution_policy_v = oneapi::dpl::execution::is_execution_policy<T>::value;
+
+      }
+
+    }
+  }
+
+``oneapi::dpl::is_execution_policy`` and ``oneapi::dpl::execution::is_execution_policy`` must be treated as name aliases
+to the same class template. It is unspecified in which namespace the underlying class template and its specializations
+are defined.
+
+``is_execution_policy<T>`` must have the characteristics of ``std::true_type`` if ``T`` is one of the above specified
+or implementation-defined execution policy types, otherwise it must have the characteristics of ``std::false_type``.
+Following the C++ Standard, ``is_execution_policy<T>`` does not automatically strip references and cv-qualifiers 
+from its template argument and should be used with ``std::decay<T>`` or similar type transformation utilities.
```

Comment:
```suggestion
from its template argument.
```
It not a direct suggestion, it's a question. Is it a part of the spec? I would just drop it, honestly

Reply:
> Is it a part of the spec?

Which spec do you mean? If C++ - no, there is nothing said about stripping references etc.
My intent is to give some usage hint for `is_execution_policy` as it might not be obvious that in practice it requires `decay`, `remove_cvref` etc. to work correctly.